static/js/click_handlers.js

```diff
@@ -547,6 +548,34 @@ export function initialize() {
             onHidden: (instance) => {
                 instance.destroy();
             },
+            onShow: (instance) => {
+                // For both buddy list and top left corner pm list, `target_node`
+                // is their parent `ul` element. We canot use MutationObserver
+                // directly on the reference elmenet because it will be removed
+                // and we need to attach it on an elment which will remain in the
+                // DOM which is their parent `ul`.
+                const target_node = instance.reference.parentElement.parentElement;
+                // We only need to know if any of the `li` elements were removed.
+                const config = {attributes: false, childList: true, subtree: false};
+                const callback = function (mutationsList) {
+                    for (const mutation of mutationsList) {
+                        // Hide instance if reference is in the removed
+                        // node list.
+                        if (
+                            [...mutation.removedNodes].some(
+                                (node) => $(node).has(instance.reference).length !== 0,
+                            )
+                        ) {
+                            instance.hide();
+                        }
+                    }
+                };
+                observer = new MutationObserver(callback);
+                observer.observe(target_node, config);
```

Comment:
@andersk can you review the MutationObserver logic here?  I can think of a few types of subtle bugs to check for:

* Leaking several observers through showing multiple tooltips over time?
* The observers being slow (I think this is controlled by only observing the `target_node`)
* A refactor in our rendering logic resulting in the parent `ul` being rewritten breaking this logic?

Also, should we move this mutation observer logic to a function in another file?  Either `buddy_list.js` or `tippyjs.js`?

Reply:
Actually, since this is being used by both buddy list and pm list tooltips, `buddy_list` is not a good place. `tippyjs` would be a better place.

Reply:
I will wait for review from @andersk before moving it though.

Reply:
I think it's reasonable to just move it where you think it belongs.

Reply:
I wonder if it would be simpler to just maintain `li_tippys: WeakMap<HTMLLIElement, Tippy>` and have `BuddyList#maybe_remove_key` invoke `li_tippys.get(li)?.hide()` directly?

Reply:
There are actually many ways the user list in the right sidebar can be updated, UserSearch, BuddyList, activity, etc.